### ISSUE TITLE
Bug: Exception encountered while parsing CSR

### ISSUE THREAD

**Issue by sfuser16** [PeculiarVentures/PKI.js#92]
Hi @YuryStrozhevsky / @rmhrisk ,

I was trying to run some scenarios and encountered an issue with a CSR parsing. I tried it on https://pkijs.org/examples/PKCS10_complex_example.html . I thought that my CSR generation process might have some issues so I ran this against different CSR parser available online and they seem to work fine everywhere else so thought to report this issue.

Here is the CSR:
-----BEGIN NEW CERTIFICATE REQUEST-----
MIIC9DCCAdwCAQAwfzELMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcT
DVNhbiBGcmFuY2lzY28xFDASBgNVBAoTC1NvbWVvbmVUZXN0MRAwDgYDVQQLEwdUZXN0aW5nMRsw
GQYDVQQDExJ0ZXN0Lm15LmRvbWFpbi5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
AQCY8P/FdgSbHbYp10XlLfsU04/tdZEgnZbPOAGeDGQrpuKeip/yA1RRE8YJdg9SsiIlyg/hZYlu
dbf0RmlzYe/KN6pjPw4A3iUXB5W6DZWWPWzY8R3y4k0OQkB+kx8OWKilGKcQcTq+pfsrbzfpGUjy
JkKq9Rg8y0K9AyfPqmK3k6cQ9SJfEszmNQ5o+Cip64GBug9aNwFX6bB12Sm41CpxqJs83OMxwmyx
+5KQ+8LLopXPSHxMDov5tM9qbL70AaSP1d3B/SIG6ZS4e39oVhfnuTocm7Ll7u6+a6d2Jfh13vl1
RVu/LBPTYh2pjaPC1RXCqfucCT+YQPPyW8B8cMSfAgMBAAGgMDAuBgkqhkiG9w0BCQ4xITAfMB0G
A1UdDgQWBBQshdy+zgHpjouDVrpdz17abcglPjANBgkqhkiG9w0BAQsFAAOCAQEAgeIIGWHytjE1
U5epSggcDTXa2VG2mKTbAZg9PQSWlIFsf1MK9IaRhTGqBO0NtV/FtKEvW0PJ0b0M9GiLBfOdzsx6
x5m9riT0PFHG8PPF4LmyuphvMJK3ozoAd3paLBgD3w21dSpTr2qyu+xtnPIuTVYvQmA6KskXDdaU
LP8Hy9+QOX9boTlhHvxj/l3qcIl3Ey5hKqr9/u3+h6/Gltn/iUPL1cbqjt+qLD1Y7ABwfe8uq3MH
JkozyTXFxRezS0x64WGXvIZ4Z930UDcEdDT34nzW1P4fmDzfXDM3T0vMZRVChEuk//zGBq/vVuDz
uz6SgLgTCxgzspJyyxYp4bGtFw==
-----END NEW CERTIFICATE REQUEST-----

The CSR was created with https://www.digicert.com/easy-csr/keytool.htm with the command run on Mac workstation and ran against different online tools available and it succeeded everywhere else.
1) http://lapo.it/asn1js/
2) https://cryptoreport.websecurity.symantec.com/checker/views/csrCheck.jsp
3) https://kjur.github.io/jsrsasign/tool_asn1dumper.html

Could you please advice if I'm doing anything wrong here?

**Comment by YuryStrozhevsky** [comment]
@sfuser16 The "problem" is in header and footer. The example for PKI.js expect these values:
```
-----BEGIN CERTIFICATE REQUEST-----
...
-----END CERTIFICATE REQUEST-----
```
And you have:
```
-----BEGIN NEW CERTIFICATE REQUEST-----
...
-----END NEW CERTIFICATE REQUEST-----
```
I will fix this in the example, but at the mean time you could yourself easily fix the "problem" in your real application.

**Comment by sfuser16** [comment]
Thanks @YuryStrozhevsky . I had already handled that "NEW" with an updated regex. I realized that it was happening because I copied the CSR content from a webpage and Mac's "copy" feature was probably entering some hidden characters when I pasted the content in the text area in the PKIjs parse csr page. Later, on opening the file with cat command in terminal and copying from there gave no errors. Should I face any other issues, I shall reach out to you. Thanks for looking into it !
